core.py

```diff
@@ -569,29 +571,32 @@ def photon_afterpulse(self, signal_pulse):
         """
         for element in self.element_list:
             delaytime_cdf = self.uniform_to_pmt_ap[element]['delaytime_cdf']
-            amplitude_cdf = self.uniform_to_pmt_ap[element]['delaytime_cdf']
+            amplitude_cdf = self.uniform_to_pmt_ap[element]['amplitude_cdf']
 
-            # Assign each photon two random uniform number rU0, rU1 from (0, 1]
+            # Assign each photon FRIST random uniform number rU0 from (0, 1] for timing
             rU0 = 1 - np.random.uniform(size=len(signal_pulse._photon_timings))
-            rU1 = 1 - np.random.uniform(size=len(signal_pulse._photon_timings))
 
             # Select those photons with U <= max of cdf of specific channel
             sel_photon_id = np.where(rU0 <= delaytime_cdf[signal_pulse._photon_channels, -1])[0]
             sel_photon_channel = signal_pulse._photon_channels[sel_photon_id]
 
-            # The map is made so that the indices are delay time in unit of ns
-            ap_delay = np.argmin(
-                np.abs(
-                    delaytime_cdf[sel_photon_channel]
-                    - rU0[sel_photon_id][:, None]), axis=-1)
+            # Assign selected photon SECOND random uniform number rU1 from (0, 1] for amplitude
+            rU1 = 1 - np.random.uniform(size=len(sel_photon_id))
 
-            if element == 'Uniform':
+            # The map is made so that the indices are delay time in unit of ns
```

Comment:
The uniform (in time) pmt afterpulses can be randomly sample directly, skip the cdf part speeds things up a lot.